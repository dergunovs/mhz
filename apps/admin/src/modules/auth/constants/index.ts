export const API_POST_LOGIN = 'auth/login';
export const API_GET_CHECK_AUTH = 'auth/check';

export const URL_LOGIN = '/';
