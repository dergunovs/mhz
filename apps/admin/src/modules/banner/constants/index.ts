export const URL_BANNER = '/banners';
export const URL_BANNER_CREATE = `${URL_BANNER}/create`;
export const URL_BANNER_EDIT = `${URL_BANNER}/edit`;
