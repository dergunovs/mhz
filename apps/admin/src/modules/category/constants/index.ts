export const URL_CATEGORY = '/categories';
export const URL_CATEGORY_CREATE = `${URL_CATEGORY}/create`;
export const URL_CATEGORY_EDIT = `${URL_CATEGORY}/edit`;

export const CATEGORY_FIELD_TYPE_OPTIONS = ['string', 'number', 'boolean'];
