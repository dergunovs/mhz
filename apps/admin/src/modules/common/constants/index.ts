export const URL_MAIN = '/main';
export const URL_ERROR = '/404';
