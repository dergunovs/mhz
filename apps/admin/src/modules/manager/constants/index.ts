export const URL_MANAGER = '/managers';
