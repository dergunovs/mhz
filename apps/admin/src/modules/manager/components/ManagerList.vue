<template>
  <UiTable
    :headers="tableHeaders"
    :isLoading="!props.managers?.length"
    :modelValue="props.modelValue"
    @update:modelValue="(value: ISortOption) => emit('update:modelValue', value)"
    @reset="(value: string) => emit('reset', value)"
  >
    <template v-if="props.managers?.length">
      <tr v-for="manager in props.managers" :key="manager._id">
        <td data-grow>
          <RouterLink :to="`${URL_MANAGER_EDIT}/${manager._id}`">
            {{ manager.email }}
          </RouterLink>
        </td>
        <td data-no-wrap>{{ manager.firstName }} {{ manager.lastName }}</td>
        <td data-no-wrap>
          {{ formatDate(manager.dateCreated) }}
        </td>
        <td data-no-wrap>
          {{ formatDate(manager.dateUpdated) }}
        </td>
      </tr>
    </template>
  </UiTable>
</template>

<script setup lang="ts">
import { IManager } from 'mhz-contracts';
import { UiTable } from 'mhz-ui';
import { formatDate, ISortOption } from 'mhz-helpers';

import { URL_MANAGER_EDIT } from '@/manager/constants';

interface IProps {
  managers?: IManager[];
  modelValue?: ISortOption;
}

const props = defineProps<IProps>();
const emit = defineEmits(['update:modelValue', 'reset']);

const tableHeaders = [
  { value: 'email', title: 'Email' },
  { value: 'lastName', title: 'Name' },
  { value: 'dateCreated', title: 'Created' },
  { value: 'dateUpdated', title: 'Updated' },
];
</script>
