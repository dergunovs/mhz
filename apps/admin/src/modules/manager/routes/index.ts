import { URL_MANAGER, URL_MANAGER_CREATE, URL_MANAGER_EDIT } from '@/manager/constants';

export const managerRoutes = [
  {
    path: URL_MANAGER,
    name: 'ManagerList',
    component: () => import('@/manager/pages/ManagerListPage.vue'),
  },
  {
    path: URL_MANAGER_CREATE,
    name: 'ManagerCreate',
    component: () => import('@/manager/pages/ManagerCreatePage.vue'),
  },
  {
    path: `${URL_MANAGER_EDIT}/:manager`,
    name: 'ManagerEdit',
    component: () => import('@/manager/pages/ManagerEditPage.vue'),
  },
];
