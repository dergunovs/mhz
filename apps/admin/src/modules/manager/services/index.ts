import { ComputedRef, Ref } from 'vue';

import { IManager } from 'mhz-types';
import { api, useMutation, useQuery, IPageQuery } from 'mhz-helpers';

import { API_MANAGER } from '@/manager/constants';

export function getManagers(query: Ref<IPageQuery | number>) {
  async function fn(): Promise<{ data: IManager[]; total: number }> {
    const params =
      typeof query.value === 'number'
        ? { page: query.value }
        : {
            page: query.value.page || 1,
            sort: query.value.sort.value,
            dir: query.value.sort.isAsc === false ? 'desc' : 'asc',
          };

    const { data } = await api.get(API_MANAGER, { params });

    return data;
  }

  return useQuery({ queryKey: [API_MANAGER, query], queryFn: fn });
}

export function getManager(id?: ComputedRef<string | string[]>) {
  async function fn(): Promise<IManager | null> {
    if (!id?.value) return null;

    const { data } = await api.get(`${API_MANAGER}/${id.value}`);

    return data;
  }

  return useQuery({ queryKey: [API_MANAGER, id], queryFn: fn, refetchOnMount: true });
}

export function postManager(options: object) {
  async function fn(formData: IManager) {
    await api.post(API_MANAGER, formData);
  }

  return useMutation({ mutationKey: [API_MANAGER], mutationFn: fn, ...options });
}

export function updateManager(id: ComputedRef<string | undefined>, options: object) {
  async function fn(formData: IManager) {
    await api.patch(`${API_MANAGER}/${id.value}`, formData);
  }

  return useMutation({ mutationKey: [API_MANAGER, id], mutationFn: fn, ...options });
}

export function deleteManager(options: object) {
  async function fn(id?: string) {
    await api.delete(`${API_MANAGER}/${id}`);
  }

  return useMutation({ mutationKey: [API_MANAGER], mutationFn: fn, ...options });
}
