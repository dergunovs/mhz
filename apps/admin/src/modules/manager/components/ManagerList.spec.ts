import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';

import ManagerList from './ManagerList.vue';

import { wrapperFactory } from '@/common/test';
import { URL_MANAGER_EDIT } from '@/manager/constants';
import { MANAGERS } from '@/manager/fixtures';

const managerListRow = '[data-test="manager-list-row"]';
const managerListLink = '[data-test="manager-list-link"]';

let wrapper: VueWrapper;

beforeEach(() => {
  wrapper = wrapperFactory(ManagerList, {
    props: {
      managers: MANAGERS.data,
    },
  });
});

enableAutoUnmount(afterEach);

describe('ManagerList', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(ManagerList)).toBeTruthy();
  });

  it('shows rows if managers props', async () => {
    expect(wrapper.findAll(managerListRow).length).toEqual(MANAGERS.data.length);

    await wrapper.setProps({ managers: [] });

    expect(wrapper.find(managerListRow).exists()).toBe(false);
  });

  it('generated link to manager', async () => {
    const LINK = `${URL_MANAGER_EDIT}/${MANAGERS.data[0]._id}`;

    expect(wrapper.find(managerListLink).attributes('to')).toEqual(LINK);
  });

  it('shows manager content in cells', async () => {
    expect(wrapper.find(managerListLink).text()).toEqual(MANAGERS.data[0].email);
  });
});
