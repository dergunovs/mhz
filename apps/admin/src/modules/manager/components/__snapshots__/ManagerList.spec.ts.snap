// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ManagerList > matches snapshot 1`] = `
"<table headers="[object Object],[object Object],[object Object],[object Object]" isloading="false" lang="en">
  <tbody>
    <tr data-test="manager-list-row">
      <td data-grow=""><a to="/managers/edit/64f19fc5677cb758200ab511" data-test="manager-list-link">a@mail.ru</a></td>
      <td data-no-wrap="">Alexandr Petrov</td>
      <td data-no-wrap="">Sep 1, 2023</td>
      <td data-no-wrap="">Sep 19, 2023</td>
    </tr>
    <tr data-test="manager-list-row">
      <td data-grow=""><a to="/managers/edit/64f19fc5677cb758200ab511" data-test="manager-list-link">a@b.ru</a></td>
      <td data-no-wrap="">Ivan Sidorov</td>
      <td data-no-wrap="">Sep 1, 2023</td>
      <td data-no-wrap="">Sep 19, 2023</td>
    </tr>
  </tbody>
</table>"
`;
